### PR TITLE
Factory pattern implemented for Enemy Creation

### PR DESCRIPTION
**Pull request by arjovanramshorst**

### CODE REVIEW

File: core/src/main/java/sem/group15/bubblebobble/core/LevelParser.java [hunk 1]
```diff
@@ -28,9 +31,25 @@
...
+                    enemyFactory = new SimpleEnemyFactory();
+                    break;
+                case "StrongEnemy":
+                    enemyFactory = new StrongEnemyFactory();
+                    break;
+                default:
+                    enemyFactory = new SimpleEnemyFactory(); // default to simple enemies, if there is no line present.
+                    result.add(getObject(enemyType)); // Get first line as object anyway.
+            }
+        }
```
Comment (DaanVermunt) [inline core/src/main/java/sem/group15/bubblebobble/core/LevelParser.java]: Clean solution for new enemy types

File: core/src/main/java/sem/group15/bubblebobble/core/objects/Enemy.java [hunk 2]
```diff
@@ -27,14 +27,7 @@
...
 
-        //Generates random integer between 1 and 2.
-        int random = 1 + (int) (Math.random() * ((2 - 1) + 1));
-        assert (random == 1 || random == 2);
-        if (random == 1) {
-            setDirection(Direction.RIGHT);
-        } else {
-            setDirection(Direction.LEFT);
-        }
+
```
Comment (DaanVermunt) [inline core/src/main/java/sem/group15/bubblebobble/core/objects/Enemy.java]: Why did you delete this part?

**Comment by DaanVermunt** [comment]
Factories look good, just need to fix merge conflict and were good to go

File: core/src/main/java/sem/group15/bubblebobble/core/objects/Enemy.java (hunk 2 above)
Comment (arjovanramshorst) [inline core/src/main/java/sem/group15/bubblebobble/core/objects/Enemy.java]: This is unnecessary construction code, it's better to handle this in the factory.

**Comment by MatthijsRijlaarsdam** [comment]
Looks excellent, we'll change the filled bubble implementation to implement this in a later stage. merging. <feature
